Documentation/users/tectonic-identity-config.md

```diff
@@ -0,0 +1,90 @@
+
+Tectonic Identity pulls its configuration options from a config file stored in a `ConfigMap`, which admins can view and edit using `kubectl`. As a precaution in case of misconfiguration, use the administrative kubeconfig in the [downloaded `assets.zip`][assets-zip] when editing Identity's config.
+
+First, use `kubectl` to backup the existing config:
+
+```
+kubectl get configmaps tectonic-identity --namespace=tectonic-system -o yaml > identity-config.yaml.bak
+```
+
+Then, edit the current `ConfigMap` with the desired changes:
+
+```
+kubectl edit configmaps tectonic-identity --namespace=tectonic-system
+```
+
+Finally, use `kubectl` to trigger  a rolling update:
+
+```
+kubectl patch deployment tectonic-identity \
+    --patch "{\"spec\":{\"template\":{\"metadata\":{\"annotations\":{\"date\":\"`date +'%s'`\"}}}}}" \
+    --namespace tectonic-system
+```
+
+Identity's deployment is intended to be resilient against invalid config files. Verify the new state to confirm the update, and restore the `ConfigMap` backup if problems are encountered.
+
+The update's success can be inspected by watching the pods in the `tectonic-system` namespace.
+
+```
+kubectl get pods --namespace=tectonic-system
+```
+
+## Add a static user
+
+Static users are those defined directly in the Tectonic Identity `ConfigMap`. Static users are intended to be used for initial setup, and may also be used for troubleshooting and recovery. A static user acts as a stand-in, authenticating users without a connection to a backend identity provider.
+
+To add a static user, update the Tectonic Identity `ConfigMap` with a new `staticPasswords` entry.
+
+```yaml
+staticPasswords:
+# The following fields are required.
+- email: "test1@example.com"
+  hash: "$2a$10$2b2cU8CPhOTaGrs1HRQuAueS7JTT5ZHsHSzYiFPm1leZck7Mc8T4W"
+    # bcrypt hash for string "password"
+  username: "test1" # username to display. NOT used during login.
+  userID: "1d55c7c4-a76d-4d74-a257-31170f2c4845"
+    # randomly generated using uuidgen.
+```
+
+A bcrypt encoded hash of the user's password can be generated using the [coreos/bcrypt-tool](https://github.com/coreos/bcrypt-tool/releases/tag/v1.0.0).
+
+When generating Tectonic Console passwords with `bcrypt-tool`, using values higher than the default of `-cost=10` may result in timeouts. bcrypt also imposes a maximum password length of 56 bytes.
+
+To ensure the static user has been added successfully, log in to Tectonic Console using the static user's username and password.
+
+## Change password for static user
+
+To change the password of an existing user, generate a bcrypt hash for the new password (using [coreos/bcrypt-tool](https://github.com/coreos/bcrypt-tool/releases/tag/v1.0.0)) and add this value to the Tectonic Identity `ConfigMap` for the selected user.
+
+```yaml
+staticPasswords:
+# Existing user entry.
+- email: "test1@example.com"
+  hash: "$2a$10$TcWtvcw0N8.xK8nKdBw80uzYij6cJwuQhtfYnEf/hEC9bRTzlWdIq"
+    # newly generated bcrypt hash
```

Comment:
This comment is slightly confusing. Maybe add it to the line above and re-align it?